plugin/trino-hive/src/main/java/io/trino/plugin/hive/ion/IonFileWriterFactory.java

```diff
@@ -95,12 +100,17 @@ public Optional<FileWriter> createFileWriter(
                     .mapToObj(ordinal -> new Column(fileColumnNames.get(ordinal), fileColumnTypes.get(ordinal), ordinal))
                     .toList();
 
+            IonWriterBuilder writerBuilder = getIonEncoding(schema).equals(TEXT_ENCODING)
+                    ? IonTextWriterBuilder.minimal()
+                    : IonBinaryWriterBuilder.standard();
+
```

Comment:
Can we just pass the encoding flag to `IonFileWriter` and let it create the builder object instead of passing it from here?

Reply:
Yeah, we can pass the flag and create the writers in `IonFileWrite`. It sounds better since it will reduce the unnecessary builder creation process.